affine/__init__.py

```diff
@@ -511,9 +511,9 @@ def __mul__(self, other):
         else:
             try:
                 vx, vy = other
-            except Exception:
+                return (vx * sa + vy * sb + sc, vx * sd + vy * se + sf)
+            except (ValueError, TypeError):
```

Comment:
Proposing to return NotImplemented only for these two exceptions, other exceptions could mean a different affine bug.